src/Things/Structures/Structure.cpp

```diff
@@ -178,10 +182,33 @@ void Structure::activate()
 }
 
 
+bool Structure::disabledByIntegrity() const
+{
+	return mIntegrity < 35;
+}
+
+bool Structure::destroyedByIntegrity() const
+{
+	float destroyChance = 1.0f - mIntegrity / 25.0f;
+	// uniform_real_distributions are in a half-closed range [lower, upper).
+	// std::nextafter allows the range to be [lower, upper]
+	std::uniform_real_distribution<double> dist(0.0, std::nextafter(1.0, 2.0));
```

Comment:
I can't help but think there must be a simpler more straightforward way to do this.

Reply:
This function fulfills the requirement:
 - Logic sets a destroyed state based on a chance when integrity is lower than 25%. The lower the integrity, the higher the chance of the building collapsing.

As the value gets closer to `0.0` the chance to destroy gets closer to `1.0`, giving the RNG more of the range to pick from and a higher chance to return `true`, which will set the state of the structure to destroy during `update`.

Reply:
The issue with `uniform_real_distribution` being a half-open range is "because the standard says so." Can't be helped unfortunately. :/

Reply:
However, the algorithm is wrong. Should be
```
float destroyChance = 1.0f - mIntegrity / 100.0f;
std::uniform_real_distribution<double> dist(0.0, std::nextafter(1.0, 2.0));
return dist(generator) < destroyChance;
```

Reply:
Opted for `float destroyChance = mIntegrity <= 25 ? 1.0f - mIntegrity / 25.0f : 0.0f;` instead.
This way the structures can only destroy by decay if they are actually at 25 or less integrity as the spec requires. The other algorithm above makes it so there's a chance for a perfectly healthy 99% integrity building to destroy itself, albeit a 1% chance.

Maybe that can be a feature when difficulty is implemented. 😈

Reply:
The half open range might not actually be a problem. The difference should amount to whether 25% integrity means the first point at which a structure can collapse, or the last point at which a structure doesn't collapse. As the point was arbitrary to begin with, the distinction is quite arbitrary. I'm kind of thinking the `std::nextafter` could just be dropped.

With the ternary, the `std::nextafter` difference amounts to the different between `<=` or strict `<`.

----

With the new formula, you might as well use `<`. The result is the same, though it avoids generating random numbers when the probability of collapse is 0.

----

Was also thinking the labelling of `destroyChance` looked a bit backwards before. Not something I'm quite worried about at this exact moment. The details of the formula used is subject to change over time.

Reply:
The `Population` code generates integers in the range `0` to `100`:
```cpp
std::uniform_int_distribution<int> pop_distribution(0, 100);
```

Perhaps we should do similar for the percent chance a building will be destroyed. It would be easy enough to convert the formula to use percentage instead of a `float` value:
```cpp
int destroyChance = mIntegrity < 25 ? 100 - (mIntegrity * 100 / 25) : 0;
```